Assets/MixedRealityToolkit.Tests/PlayModeTests/PressableButtonTests.cs

```diff
@@ -0,0 +1,237 @@
+            Assert.IsNotNull(buttonComponent);
+
+            testButton.transform.Translate(new Vector3(10.0f, 5.0f, 20.0f));
+
+            PressableButton.SpaceMode distanceMode = buttonComponent.DistanceSpaceMode;
+            // check default value -> default must be using world space to not introduce a breaking change to the button
+            Assert.IsTrue(distanceMode == PressableButton.SpaceMode.World, "Pressable button default value is using local space distances which introduces a breaking change for existing projects");
+
+            // make sure there's no scale on our button
+            testButton.transform.localScale = new Vector3(1.0f, 1.0f, 1.0f);
+
+            // set start distance -> default is zero
+            buttonComponent.StartPushDistance = 0.00003f;
+
+            // get the buttons default values for the push planes
+            float startPushDistance = buttonComponent.StartPushDistance;
+            float maxPushDistance = buttonComponent.MaxPushDistance;
+            float pressDistance = buttonComponent.PressDistance;
+            float releaseDistance = pressDistance - buttonComponent.ReleaseDistanceDelta;
+
+            Vector3 startPushDistanceWorld = buttonComponent.GetWorldPositionAlongPushDirection(startPushDistance);
+            Vector3 maxPushDistanceWorld = buttonComponent.GetWorldPositionAlongPushDirection(maxPushDistance);
+            Vector3 pressDistanceWorld = buttonComponent.GetWorldPositionAlongPushDirection(pressDistance);
+            Vector3 releaseDistanceWorld = buttonComponent.GetWorldPositionAlongPushDirection(releaseDistance);
+
+            // scale the button in z direction
+            // scaling the button while in world space shouldn't influence our button plane distances
+            testButton.transform.localScale = new Vector3(1.0f, 1.0f, 2.0f);
+
+            Vector3 startPushDistanceWorldScaled = buttonComponent.GetWorldPositionAlongPushDirection(startPushDistance);
+            Vector3 maxPushDistanceWorldScaled = buttonComponent.GetWorldPositionAlongPushDirection(maxPushDistance);
+            Vector3 pressDistanceWorldScaled = buttonComponent.GetWorldPositionAlongPushDirection(pressDistance);
+            Vector3 releaseDistanceWorldScaled = buttonComponent.GetWorldPositionAlongPushDirection(releaseDistance);
+
+            // compare our distances
+            Assert.IsTrue(startPushDistanceWorld == startPushDistanceWorldScaled, "Start Distance was modified while scaling button gameobject");
+            Assert.IsTrue(maxPushDistanceWorld == maxPushDistanceWorldScaled, "Max Push Distance was modified while scaling button gameobject");
+            Assert.IsTrue(pressDistanceWorld == pressDistanceWorldScaled, "Press Distance was modified while scaling button gameobject");
+            Assert.IsTrue(releaseDistanceWorld == releaseDistanceWorldScaled, "Release Distance was modified while scaling button gameobject");
```

Comment:
Since this is actually not relying on running the editor / doing hand movements / getting focus events, could this actually be an editor test?

Reply:
I suppose it could be - the reason I added it as a runtime test was that we had some customers scaling their buttons during runtime, so i wanted to simulate that scenario - in the end it shouldn't matter anymore as the calculations are the same now